### PR TITLE
Bugfix/2156

### LINKED ISSUE #2156: string out of bounds on getPathsInPath
## Exception
* __User Action:__ UI Error
* __Request:__ Application crash
* __Version:__ 3.5.2
* __OS:__ Linux Android 10 - 29
* __Device:__ picasso
* __Model:__ Redmi K30 5G
* __Product:__ picasso
<details><summary><b>Crash log </b></summary><p>

```
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
at java.lang.String.substring(String.java:2064)
at com.amaze.filemanager.filesystem.files.FileUtils.getPathsInPath(FileUtils.java:586)
at com.amaze.filemanager.ui.views.appbar.BottomBar.showButtons(BottomBar.java:238)
at com.amaze.filemanager.ui.views.appbar.BottomBar.updatePath(BottomBar.java:552)
at com.amaze.filemanager.ui.fragments.TabFragment.updateBottomBar(TabFragment.java:416)
at com.amaze.filemanager.ui.fragments.TabFragment.updatepaths(TabFragment.java:198)
at com.amaze.filemanager.ui.activities.MainActivity.updatePaths(MainActivity.java:1299)
at com.amaze.filemanager.ui.fragments.MainFragment.reloadListElements(MainFragment.java:1088)
at com.amaze.filemanager.ui.fragments.MainFragment.setListElements(MainFragment.java:1002)
at com.amaze.filemanager.ui.fragments.MainFragment.lambda$loadlist$3(MainFragment.java:954)
at com.amaze.filemanager.ui.fragments.MainFragment.lambda$loadlist$3$MainFragment(Unknown Source:0)
at com.amaze.filemanager.ui.fragments.-$$Lambda$MainFragment$9qQv21X8JwPlLXi_cDXfu4p-_VI.onAsyncTaskFinished(Unknown Source:8)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.onPostExecute(LoadFilesListTask.java:269)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.onPostExecute(LoadFilesListTask.java:67)
at android.os.AsyncTask.finish(AsyncTask.java:755)
at android.os.AsyncTask.access$900(AsyncTask.java:192)
at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:772)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loop(Looper.java:224)
at android.app.ActivityThread.main(ActivityThread.java:7584)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)

```
</details>
<hr>

### PR DESCRIPTION
**Pull request by VishalNehra**
#### Issue tracker   
Fixes will automatically close the related issue

Fixes #2156 
-or-   
Addresses #

#### Release  
Addresses release/
  
#### Test cases
- [ ] Covered
  
#### Manual testing
- [x] Done  
  
If yes,  
- Device: Oneplus 6t
- OS: Android 10

#### Build tasks success  
Successfully running following tasks on local 
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

#### Related PR  
Related to PR #2228 

#### Additional Info
Also enable pressing on appbar for all open modes, so can jump multiple paths now

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java [hunk 1]
```diff
@@ -118,7 +119,9 @@ public LoadFilesListTask(
         if (hFile == null) {
           hFile = new HybridFile(OpenMode.SMB, path);
         }
-
+        if (!hFile.getPath().endsWith("/")) {
+          hFile.setPath(hFile.getPath() + "/");
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:123]: Why is this needed?

Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:123]: Need to append a training slash for SMB file loading if one is not present at the end of path. I had raised this point on telegram. Logic from AppBar button presses doesn't have one at the end.

Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:123]: Shouldn't it be done before in the execution?

Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:123]: We only have open mode switch in this asynctask, before this we don't modify path based on open modes. 
Plus it's not good idea to write such logic in Fragment side.

**Comment by EmmanuelMess** [comment]
Added two small fixes

**Review by EmmanuelMess — APPROVED** [review]

**Review by TranceLove — APPROVED** [review]

**Pull request merged**